static/js/compose.js

```diff
@@ -471,7 +471,9 @@ exports.validation_error = function (error_type, stream_name) {
         compose_error(i18n.t("Error checking subscription"), $("#stream"));
         return false;
     case "not-subscribed":
-        var new_row = templates.render("compose_not_subscribed");
+        var new_row = templates.render("compose_not_subscribed", {
+            should_display_sub_button: !page_params.is_guest &&
+                                       !stream_data.get_invite_only(stream_name)});
```

Comment:
Would it be cleaner to do `stream_data.get()` and then `sub.should_display_subscription_button` ?

Reply:
yes, in future also if we change flag's definition. Changed!